src/librustdoc/passes/collect_intra_doc_links.rs

```diff
@@ -986,6 +986,9 @@ impl LinkCollector<'_, '_> {
                 }
             }
 
+            let stripped_path_string = strip_generic_parameters(path_str);
+            path_str = &stripped_path_string;
```

Comment:
This allocates a new copy of each link. Can you change it to only allocate if there are generics present?
```suggestion
            if path_str.contains(&['<', '>']) {
                stripped_path_string = strip_generic_parameters(path_str);
                path_str = &stripped_path_string;
            }
```